src/wallet/walletdb.h

```diff
@@ -240,11 +246,19 @@ class WalletBatch
 
     bool WriteMinVersion(int nVersion);
 
+    bool WriteDescriptorKey(const uint256& desc_id, const CPubKey& pubkey, const CPrivKey& privkey);
+    bool WriteCryptedDescriptorKey(const uint256& desc_id, const CPubKey& pubkey, const std::vector<unsigned char>& secret);
+    bool WriteDescriptor(const uint256& desc_id, const WalletDescriptor& descriptor);
+    bool WriteDescriptorDerivedCache(const uint256& desc_id, uint32_t key_exp_index, uint32_t der_index, const CExtPubKey& xpub);
```

Comment:
Perhaps put `uint32_t der_index` last, so the first two parameters are the same as in `WriteDescriptorParentCache()` on the following line

Reply:
I moved `xpub` to the front so that the only difference is that `WriteDescriptorDerivedCache` has `der_index` at the end.